saml2/saml2-service-provider/src/main/java/org/springframework/security/saml2/provider/service/registration/Saml2MessageBinding.java

```diff
@@ -26,13 +32,21 @@
  */
 public enum Saml2MessageBinding {
 
-	POST("urn:oasis:names:tc:SAML:2.0:bindings:HTTP-POST"), REDIRECT(
-			"urn:oasis:names:tc:SAML:2.0:bindings:HTTP-Redirect");
+	POST("urn:oasis:names:tc:SAML:2.0:bindings:HTTP-POST", (request) -> "POST".equalsIgnoreCase(request.getMethod())
```

Comment:
We cannot do this work here since `HttpServletRequest` references are restricted to the `web` package.

Additionally, Spring Security prefers to place behavior as much as possible into components that implement interfaces. Generally, this simplifies applications overriding the behavior.